tuf/api/metadata.py

```diff
@@ -1219,6 +1219,74 @@ def to_dict(self) -> Dict[str, Any]:
             **self.unrecognized_fields,
         }
 
+    @classmethod
+    def from_file(
+        cls, filepath: str, hash_algorithms: Optional[List[str]] = None
+    ) -> "TargetFile":
```

Comment:
This method still needs the target path argument that from_data() already has:
* filepath is the local path to the content we should hash
* then additionally we need a target path (which is a URL path-part decided by caller) that ends up as `TargetFile.path`. Currently code use the local path but that's not necessarily correct.

The naming should be consistent so that both methods use the same argument name for target path. Maybe this methods arguments should be `local_path` and `target_path` while from_data() has `data` and `target_path`?

The order of args should be consistent as well: so target_path is in same position for both methods.

Reply:
I've misunderstood the path argument. Thanks for clarifying that for me, @jku! I've changed the code accordingly.